src/web/mjs/connectors/CxC.mjs

```diff
@@ -0,0 +1,254 @@
+        this.requestOptions.headers.set('content-type', 'application/json;charset=utf-8');
+        this.requestOptions.headers.set('referer', this.url);
+        this.requestOptions.headers.set('origin', this.url + '/');
+        this.requestOptions.headers.set('device', 'web');
+        this.links = {
+            login: 'https://cxc.today/login'
+        };
+        this.config = {
+            username: {
+                label: 'E-Mail',
+                description: 'E-Mail for login with your CxC account.\nAn account is required to access 18+ content (18禁).\nA fan subscription is required to access VIP content (訂閱粉絲限定).',
+                input: 'text',
+                value: ''
+            },
+            password: {
+                label: 'Password',
+                description: 'Password for login with your CxC account.\nAn account is required to access 18+ content (18禁).\nA fan subscription is required to access VIP content (訂閱粉絲限定).',
+                input: 'password',
+                value: ''
+            }
+        };
+    }
+
+    async _initializeConnector() {
+        await this._getUUID();
+        await this._login();
+    }
+
+    async _getMangaFromURI(uri) {
+        await this._login();
+        const bookAddress = uri.href;
+        const storeName = bookAddress.match(/\/store\/([a-zA-Z0-9_]+)\//)[1];
+        const bookId = bookAddress.match(/\/book\/(\d+)/)[1];
+        let storeRequest = new Request(`${this.api}/store/${storeName}?v=2`, this.requestOptions);
+        let storeData = await this.fetchJSON(storeRequest);
+        const id = `/store/${storeData.data.id}/book/${bookId}`;
+        let bookRequest = new Request(this.api + id, this.requestOptions);
+        let bookData = await this.fetchJSON(bookRequest);
+        if (bookData.data.work_category.id != 1) {
+            throw Error('CxC book type must be comic (漫畫).');
+        }
+        return new Manga(this, id, bookData.data.name);
+    }
+
+    async _getMangasFromPage(page) {
+        await this._login();
+        try {
+            let request = new Request(`${this.api}/book?page=${page}&per_page=${this.booksPerPage}&sort_by=updated_at&work_category=1&lang=&word_count=0`, this.requestOptions);
+            let bookData = await this.fetchJSON(request);
+            bookData = bookData.data.data.filter(book => book.work_category.id == 1);
+            return bookData.map(book => {
+                return {
+                    id: `/store/${book.store.id}/book/${book.id}`,
+                    title: book.name
+                };
+            });
+        } catch (error) {
+            return [];
+        }
+    }
+
+    async _getMangas() {
+        await this._login();
+        let books = [];
+        for (let page = 1, run = true; run; page++) {
+            const booksFromPage = await this._getMangasFromPage(page);
+            booksFromPage.length >= this.booksPerPage ? books.push(...booksFromPage) : run = false;
```

Comment:
No. keep the check with > 0. In case last page may have less manga that requested they will be lost with this ocde